bundles/AdminBundle/Controller/Admin/DataObject/DataObjectHelperController.php

```diff
@@ -1801,16 +1803,25 @@ protected function getCsvFile($fileHandle)
      *
      * @param Request $request
      * @param GridHelperService $gridHelperService
+     * @param EventDispatcherInterface $eventDispatcher
      *
      * @return JsonResponse
      */
-    public function getExportJobsAction(Request $request, GridHelperService $gridHelperService)
+    public function getExportJobsAction(Request $request, GridHelperService $gridHelperService, EventDispatcherInterface $eventDispatcher)
     {
         $requestedLanguage = $this->extractLanguage($request);
         $allParams = array_merge($request->request->all(), $request->query->all());
 
         $list = $gridHelperService->prepareListingForGrid($allParams, $requestedLanguage, $this->getAdminUser());
 
+        $beforeListLoadEvent = new GenericEvent($this, [
+            'list' => $list,
+            'context' => $allParams,
+        ]);
+        $eventDispatcher->dispatch(AdminEvents::OBJECT_LIST_BEFORE_LIST_LOAD, $beforeListLoadEvent);
```

Comment:
I think we should introduce a new event `AdminEvents::OBJECT_LIST_BEFORE_LIST_PREPARE`, which is triggered here. then it will not change the behavior of export, if there are existing listeners/subscribers for grid listing.

Reply:
I was in doubt to do this or not, since the event is described as 'This is a global event (search list, grid list, tree list, ...).'
But for BC reasons I do agree it is better to have a separate event. I will change it and update the PR.